laser_runner_removal/laser_runner_removal/nodes/test_node.py

```diff
@@ -0,0 +1,128 @@
+import rclpy
+from simple_node import Node
+from yasmin import State, StateMachine
+from std_srvs.srv import Empty
+from laser_control_interfaces.msg import Point
+from laser_control_interfaces.srv import (
+    AddPoint,
+    ConnectToDac,
+    GetBounds,
+    Play,
+    SetColor,
+)
+import time
+
+
+class InitializationState(State):
+    """Initialization state for setting up resources needed by the system"""
+
+    def __init__(self, node):
+        State.__init__(self, outcomes=["initialization_success"])
+        self.node = node
+
+    def execute(self, blackboard):
+        self.node.get_logger().info("State: InitializationState")
+
+        # Connect to laser
+        while not self.node.laser_connect_to_dac_cli.wait_for_service(timeout_sec=1.0):
+            self.node.get_logger().info("Laser not available, waiting...")
+
+        future = self.node.laser_connect_to_dac_cli.call_async(
+            ConnectToDac.Request(idx=0)
+        )
+        rclpy.spin_until_future_complete(self.node, future)
+
+        if not future.result().success:
+            "initialization_failure"
+
+        return "initialization_success"
+
+
+class CalibrationState(State):
+    """Calibrate the laser to camera, creating a 3D pos to laser frame transform."""
+
+    def __init__(self, node):
+        State.__init__(self, outcomes=["calibration_failure", "calibration_success"])
+        self.node = node
+
+    def execute(self, blackboard):
+        self.node.get_logger().info("State: CalibrationState")
+
+        self.node.laser_set_color_cli.call_async(
+            SetColor.Request(r=1.0, g=0.0, b=0.0, i=1.0)
+        )
+
+        future = self.node.laser_get_bounds_cli.call_async(GetBounds.Request(scale=1.0))
```

Comment:
I'm realizing how annoying it is to call into services - I wish client code was automatically generated

Reply:
Yeah, there's definitely a lot of boilerplate all over that feels like it could be autogenerated. I'm also interested in the average latency between a call and the action occurring.